### DISCUSSION TITLE
Control to delay zero-rpm for an amount of time below minimum temperature

### DISCUSSION THREAD

**Discussion by paridkushta** [Rem0o/FanControl.Releases#2675] [Ideas]
I cannot seem to be able to make this work with Triggers/Mix curves, please help me out if I'm missing something.

I use linear curves from 60C to 70C, from 0% to 100%, with 1 second hysteresis (only-down), 1 second response time, not ignoring hysteresis at min/max temperatures.

My system has the capability to uphold itself with no active cooling for non-gaming tasks. Currently I am forced to put (PWM) fans to idle speed all the time, where they rotate slowly for nothing. I can switch them to DC mode and make them shut off like that.

What I want to achieve is for the fans to start spinning at 60C (from zero), then whenever the temperature falls under 60C (which is often in transitional periods in-game) the fans should stay at an idle speed (mine are 40%) for a user-set period of time, after which they go to 0%.

This could be set as an extra control inside curve cards as "zero-rpm", which when enabled shows the timer box and idle speed similarly to the trigger card. Or even better (in my opinion) to have these options in each fan control card.

This is necessary for all zero-rpm configurations because the computer keeps turning the fans on and off every second. Increasing hysteresis fixes one problem and creates a bigger one by making fans not work linearly during heavier workloads, either keeping them too fast or suddenly slowing them.

There is literally no solution in any software including BIOSes, and it seems a simple thing to implement in FanControl (not saying it's easy). 
With this option fans avoid the wear at startup and making noise because of startup. Seems like a no-brainer to me that it should be mandatory with any fan control software, in principle.

**Comment by Rem0o** [comment]
Put a trigger in the mix with that 40% fan speed at load.   Put a 10-15-20 seconds response time, whatever delay you want.  Idle fan speed 0%.

Thus the trigger will hold the load fan speed (here 40%) until it hits idle temp or below FOR AT LEAST 10-15-20 seconds.

**Reply by paridkushta** [reply]
I tried that earlier and it didn't work, i set 58C idle and 59C load temp. My logic was that at 59C it would start spinning, and at 58C it would start counting the timer (300 seconds in mine but also tried with 20 seconds) then shut the fans off.

Also, does the linear curve completely overwrite the trigger at 60+C?

**Reply by Rem0o** [reply]
The mix with the "max" function selected will just pick the highest in the mix.   Your problem is that the response time (300 here) also works on the way up. So if your load doesn't last at least 300 seconds, the trigger will never fire.   Which would kinda make sense, why would you want to cool off your system 300 seconds over a 10 second load for example.

**Reply by paridkushta** [reply]
I would want to cool my system for 300 seconds because if it makes it past 60C, it means that very soon a big load is coming (no pun intended). EDIT: Not to mention that the system will be heat-soaked by being zero-rpm beforehand.

The 300 seconds are necessary to determine if a game is still running (giving the game time to start pushing temperatures again).

I wouldn't want the fans to turn off and then wake up immediately at 100%. So there's no way to have separate response times for increases and decreases in temperature?

**Reply by Rem0o** [reply]
As of right now there is no individual response time for up and down. But in your case of gaming, wouldn't you play for at least a few minutes anyway before needing that long cooling period? Try to live a bit with like a 240 second response time on the trigger and see if that ends up working for your use case.

**Reply by paridkushta** [reply]
It works but it's not 100%. If after 300 seconds (or whatever seconds) the temperature is suddenly low (because i might be alt+tabbing out of the game to reply to messages) the fans will shut off. Why leave things to chance?

It's better than nothing but in the long term it might be worth implementing. Much simpler than setting up triggers too, for less savvy individuals.
